src/Illuminate/Http/Client/Pool.php

```diff
@@ -34,17 +34,26 @@ class Pool
      * Create a new requests pool.
      *
      * @param  \Illuminate\Http\Client\Factory|null  $factory
+     * @param  \Illuminate\Http\Client\PendingRequest|null  $pendingRequest
      * @return void
      */
-    public function __construct(Factory $factory = null)
+    public function __construct(Factory $factory = null, PendingRequest $pendingRequest = null)
```

Comment:
Is it still needed to pass in the factory when you already leverage a `PendingRequest` which was already instantiated with the factory?

It seems highly unlikely this pool will be instantiated at any time without the initial `PendingRequest`.

Reply:
I did this so it would be less likely to be considered a breaking change.

Reply:
Happy to see what the maintainer(s) think, as it would be nice to completely remove the Factory variable.